### DISCUSSION TITLE
How to use Goldwarden autotype feature in Linux KDE?

### DISCUSSION THREAD

**Discussion by MurzNN** [quexten/goldwarden#71] [Q&A] [answered]
Please help me understand how to use Goldwarden autotype feature in Linux KDE?

I went through the steps described here https://github.com/quexten/goldwarden/wiki/System%E2%80%90wide-Autotype but still can't understand how to use it on Kubuntu 23.04 (Ubuntu with KDE).

In the KDE I configured a hotkey (Meta+C) to launch the command:
```
dbus-send --type=method_call --dest=com.quexten.Goldwarden.autofill /com/quexten/Goldwarden com.quexten.Goldwarden.Autofill.autofill
```
Then I open an application with a login-password input fields, put focus on the password field.
Then, the Goldwarden window appears, where I can type the credentials name and find it.
But what to do after this?

If I click or even double-click on the credentials item - nothing happens, the window still there. The same is with "Enter" key.

If I click "P" - the "Password copied to the clipboard" message appears, but no password actually is copied to clipboard.

So, please explain how to use the autotype feature right?

**Comment by quexten** [comment]
Hmm, so hitting enter when having selected an entry should show a prompt to allow "remote desktop access" (for autotype). When you accept, it should autotype. And hitting p should copy the password properly. So there is something buggy here. 

Is your KDE session on Wayland on X11?

**Comment by MurzNN** [comment]
On X11, here is the system information:
> Operating System: Kubuntu 23.10
KDE Plasma Version: 5.27.8
KDE Frameworks Version: 5.110.0
Qt Version: 5.15.10
Kernel Version: 6.5.0-14-generic (64-bit)
Graphics Platform: X11

Seems some lib, that should show the prompt, missing?

**Reply by quexten** [reply]
Ok, seems the copy paste is not working because wl-copy is used https://github.com/quexten/goldwarden/blob/main/ui/clipboard.py ; I don't test X11 actively which is why this slipped through.

**Reply by quexten** [reply]
https://github.com/quexten/goldwarden/issues/73

**Reply by quexten** [reply]
Not entirely sure yet about the autotype part, I'll test in a vm later today or tomorrow. Could be the kde portal, could be x11.

**Reply by MurzNN** [reply]
Yes, the `wl-copy` binary is missing. For x11 you can use something like `echo "$password" | xsel  -b` or just `xsel -b` and directly put the string to the stdin.

**Reply by quexten** [reply]
Fixed on latest main by using GTK's clipboard instead.

**Reply by MurzNN** [reply]
Thanks, copy to clipboard now works well! The next great step is - autotyping it ;-)

**Comment by MurzNN** [comment]
For autotyping maybe we can reuse this package https://snyk.io/advisor/python/autotype ?

**Reply by quexten** [reply]
That could work. Alternatively, just pyautogui (since that has windows/mac support too, and supports key injection). I have not played around with the packages yet, to make a call.

https://github.com/quexten/goldwarden/issues/91

**Reply by quexten** [reply, marked as answer]
Implemented in #194
